io/src/csv/read_cells.cpp

```diff
@@ -81,106 +76,56 @@ void read_cells(
     const digitization_config* dconfig,
     const std::map<std::uint64_t, detray::geometry::barcode>* barcode_map) {
 
+    // Temporary storage for all the cells and modules.
+    std::map<std::uint64_t, std::set<traccc::cell> > cellsMap;
+
     // Construct the cell reader object.
     auto reader = make_cell_reader(filename);
 
-    // Create cell counter vector.
-    std::vector<unsigned int> cellCounts;
-    cellCounts.reserve(5000);
-
-    cell_module_collection_types::host& result_modules = out.modules;
-    result_modules.reserve(5000);
-
-    // Create a cell collection, which holds on to a flat list of all the cells
-    // and the position of their respective cell counter & module.
-    std::vector<std::pair<csv::cell, unsigned int>> allCells;
-    allCells.reserve(50000);
-
     // Read all cells from input file.
     csv::cell iocell;
+    unsigned int nduplicates = 0;
     while (reader.read(iocell)) {
 
-        // Modify the geometry ID of the cell if a barcode map is provided.
-        const std::uint64_t original_geometry_id = iocell.geometry_id;
+        // Add the cell to the module. At this point the module link of the
+        // cells is not set up correctly yet.
+        if (cellsMap[iocell.geometry_id]
+                .insert({iocell.channel0, iocell.channel1, iocell.value,
+                         iocell.timestamp, 0})
+                .second == false) {
+            ++nduplicates;
+        }
```

Comment:
As it turns out, it's not quite as easy as I thought. :frowning: Since one can't just update the elements of `std::set`-s. As that would mess with the ordering in the container.

So for now I didn't add code for summing up the activation values. :thinking: I'd really like to keep the code as simple as possible, so I'm all ears on how others would do this.

Reply:
Remember you can update the values in a map, and use the geometry-channel triplet as the key.

Reply:
You're right. I did it with an `std::map` in the end instead of an `std::set`.

So now the code also sums up the activation values from the duplicate cells. :thinking: